core/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -322,8 +483,20 @@ public static SnapshotInfo readSnapshotInfo(StreamInput in) throws IOException {
      * @param in stream input
      * @return deserialized snapshot info or null
      */
-    public static SnapshotInfo readOptionalSnapshotInfo(StreamInput in) throws IOException {
-        return in.readOptionalStreamable(SnapshotInfo::new);
+    public static SnapshotInfo readOptionalSnapshotInfo(final StreamInput in) throws IOException {
```

Comment:
And here just call `readOptionalWriteable` at the call site.

You might could also do `in.readOptionalWriteable(SnapshotInfo::new);` which is a bit more concise if it works.